Update acquire documentation with volatile information (#39)
fox-it/dissect-docs: docs/source/tools/acquire.rst

               WindowsNotifications, SSH, IIS
      linux  : Etc, Boot, Home, SSH, Var, History, WebHosting
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH, VMFS
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo, History,
               SSH

    default profile:
      windows: NTFS, EventLogs, Registry, Tasks, PowerShell,
               Prefetch, Appcompat, PCA, Misc, ETL, Recents,
               RecycleBin, Drivers, Syscache, WBEM, AV, BITS,
               DHCP, DNS, ActiveDirectory, RemoteAccess,
               ActivitiesCache
      linux  : Etc, Boot, Home, SSH, Var
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH, VMFS
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo

    minimal profile:
      windows: NTFS, EventLogs, Registry, Tasks, PowerShell,
               Prefetch, Appcompat, PCA, Misc
      linux  : Etc, Boot, Home, SSH, Var
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo

Profile ``none`` is a special case where no module gets collected.

> Profiles can be used in combination with ``--dir``, ``--file`` or ``--glob`` to collect specific paths from a target.

Reviewer: ```suggestion
Profiles can be used in combination with ``--dir``, ``--file`` or ``--glob`` to collect specific user-defined paths from a target.
```

This makes ``acquire`` an excellent tool to, among others, speed up the process of digital forensic triage.
``acquire`` uses ``dissect`` to gather forensic artifacts from the raw disk, if possible.

The most basic usage of ``acquire`` is as follows:

.. code-block:: console

    $ sudo acquire

The tool requires administrative access to read raw disk data.
However, there are some options available to use the local operating systems's filesystem as a fallback option. (e.g ``--fallback`` or ``--force-fallback``)

Filesystem acquisition
----------------------

``acquire`` gathers artifacts based on modules.
These modules contain paths or globs that ``acquire`` attempts to gather from a filesystem.

> ``acquire`` can execute multiple modules at once, where a collection of them are combined into a profile.

Reviewer 2: ```suggestion
``acquire`` can execute multiple modules at once (comma separated). Instead of specifying modules manually, it's possible to use a predefined collection known as a profile.
```

Reviewer 2: oh yes, my mistake

This makes ``acquire`` an excellent tool to, among others, speed up the process of digital forensic triage.
``acquire`` uses ``dissect`` to gather forensic artifacts from the raw disk, if possible.

The most basic usage of ``acquire`` is as follows:

.. code-block:: console

    $ sudo acquire

The tool requires administrative access to read raw disk data.
However, there are some options available to use the local operating systems's filesystem as a fallback option. (e.g ``--fallback`` or ``--force-fallback``)

Filesystem acquisition
----------------------

> ``acquire`` gathers artifacts based on modules.

Reviewer 3: ```suggestion
``acquire`` gathers artefacts based on modules.
```

Reviewer 3: Yes.

               WindowsNotifications, SSH, IIS
      linux  : Etc, Boot, Home, SSH, Var, History, WebHosting
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH, VMFS
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo, History,
               SSH

    default profile:
      windows: NTFS, EventLogs, Registry, Tasks, PowerShell,
               Prefetch, Appcompat, PCA, Misc, ETL, Recents,
               RecycleBin, Drivers, Syscache, WBEM, AV, BITS,
               DHCP, DNS, ActiveDirectory, RemoteAccess,
               ActivitiesCache
      linux  : Etc, Boot, Home, SSH, Var
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH, VMFS
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo

    minimal profile:
      windows: NTFS, EventLogs, Registry, Tasks, PowerShell,
               Prefetch, Appcompat, PCA, Misc
      linux  : Etc, Boot, Home, SSH, Var
      bsd    : Etc, Boot, Home, SSH, Var, BSD
      esxi   : Bootbanks, ESXi, SSH
      osx    : Etc, Home, Var, OSX, OSXApplicationsInfo

Profile ``none`` is a special case where no module gets collected.

> ``profiles`` can be used in combination with ``--dir``, ``--file`` or ``--glob`` to collect specific paths from a target.

Reviewer 3: Why is profiles monotyped?

You can specify these arguments multiple times for every file, directory or glob you want to collect.

Volatile acquisition
--------------------

> 

Reviewer 2: *new in Acquire 3.11*

Reviewer 2: No, such detail will only confuse

> ``acquire`` has support for volatile information collection.
> In other words, information that will be lost once a computer turns off.
> These are special modules that will only be executed at the end of all the other modules.
> 
> Acquire collects volatile windows information using this methods, which it stores in the ``$metadata$`` directory inside the acquire container.
> 
> * commands: Specific commands, either cmd or powershell.
> * ctypes: Calling windows internals with python to store information.
> 
> For linux systems, acquire has the ability to interpret the ``proc`` and ``sys`` filesystem.
> It stores this information inside their respective directory on the ``acquire`` container.
> E.g. ``/proc/1/...`` or ``/sys/fs/...``.

Reviewer 2: ```suggestion
Use ``--volatile-profile`` to obtain artifacts that are not persistent (but in RAM). Volatile Windows artifacts are stored in de ``$metadata$`` folder in the resulting archive. For Linux systems, ``proc`` and ``sys`` are acquired (saved under ``/proc/1/...`` or ``/sys/fs/...``). Windows volatile artifacts are acquired through the use of internal Windows commands and the Python c-types interface.
```

> Profiles

Reviewer 2: ```suggestion
Volatile profiles
```

^^^^^^^^

> *new in Acquire 3.11*
> 
> With the addition of volatile profiles ( used with ``--volatile-profile``) multiple volatile modules can be collected at once.
> There are currently three of these profiles ``default``, ``extensive`` and ``none``.

Reviewer 2: ```suggestion
Like regular *profiles*, the *volatile profiles* allow you to run predefined groups of modules. The following volatile profiles are available:
```

You can specify these arguments multiple times for every file, directory or glob you want to collect.

Volatile acquisition
--------------------

> Use ``--volatile-profile`` to obtain artefacts that are not persistent on disk but are located in RAM.

Reviewer: ```suggestion
Use ``--volatile-profile`` to obtain artefacts that are not persistent on disk but are located in memory.
```

Volatile Windows artefacts are stored under the ``$metadata$`` folder in the resulting archive.
Windows volatile artefacts are gathered through the use of internal Windows commands and Python's ``ctypes`` interface.
For Linux systems, ``/proc`` and ``/sys`` are gathered and stored under ``/proc/1/...`` or ``/sys/fs/...`` respectively in the resulting archive.

Volatile Profiles
^^^^^^^^^^^^^^^^^

Like regular *profiles*, the *volatile profiles* allow you to run predefined groups of volatile modules.
These profiles, are ``default``, ``extensive`` and ``none``, where ``none`` is the default.
As with ``--profile``, what gets collected depends on the detected operating system.

The following list shows the modules that belong to each ``volatile profile``.

.. code-block:: text

      default profile:
        windows: Netstat, WinProcesses, WinProcEnv, WinArpCache,
                 WinRDPSessions, WinDnsClientCache

      extensive profile:
        windows: Netstat, WinProcesses, WinProcEnv, WinArpCache,
                 WinRDPSessions, WinDnsClientCache
        linux  : Proc, Sys
        bsd    : Proc, Sys
        esxi   : Proc, Sys

> Both ``--volatile-profile`` and ``--profile`` can be used simultaneously.

Reviewer: ```suggestion
Both ``--volatile-profile`` and ``--profile`` can be used simultaneously and configured separately from each other.
```

You can specify these arguments multiple times for every file, directory or glob you want to collect.

Volatile acquisition
--------------------

> *new in Acquire 3.11*

Reviewer: I think we can leave out the mention of a version entirely.
```suggestion
```

> Use ``--volatile-profile`` to obtain artifacts that are not persistent (but in RAM).

Reviewer 3: ```suggestion
Use ``--volatile-profile`` to obtain artifacts that are not persistent on disk but are located in RAM.
```

> Volatile Windows artifacts are stored in de ``$metadata$`` folder in the resulting archive.

Reviewer: ```suggestion
Volatile Windows artefacts are stored under the ``$metadata$`` folder in the resulting archive.
```

> Windows volatile artifacts are acquired through the use of internal Windows commands and the Python c-types interface.

Reviewer 3: ```suggestion
Windows volatile artifacts are acquired through the use of internal Windows commands and the Python ``ctypes`` interface.
```

> Windows volatile artifacts are acquired through the use of internal Windows commands and the Python ``ctypes`` interface.

Reviewer: ```suggestion
Windows volatile artefacts are gathered through the use of internal Windows commands and Python's ``ctypes`` interface.
```

> For Linux systems, ``proc`` and ``sys`` are acquired (saved under ``/proc/1/...`` or ``/sys/fs/...``). 

Reviewer 3: ```suggestion
For Linux systems, ``proc`` and ``sys`` are acquired and stored under ``/proc/1/...`` or ``/sys/fs/...`` respectively. 
```

> For Linux systems, ``proc`` and ``sys`` are acquired and stored under ``/proc/1/...`` or ``/sys/fs/...`` respectively.

Reviewer: ```suggestion
For Linux systems, ``/proc`` and ``/sys`` are gathered and stored under ``/proc/1/...`` or ``/sys/fs/...`` respectively in the resulting archive.
```

Volatile Profiles
^^^^^^^^^^^^^^^^^

Like regular *profiles*, the *volatile profiles* allow you to run predefined groups of modules.
The following volatile profiles are available:

* ``default``: Collect a set of volatile information.
* ``extensive``: Collect a lot, if not all, the volatile information. Usually the modules that either take a lot of time or are very involved.

> * ``none``: Do not collect any volatile information.

Reviewer: This is the default behaviour without specifying `--volatile-profile` right? It might be a good idea to convey that here as well.

> * ``none``: Do not collect any volatile information.

Reviewer: I think it is a good idea to make the explanation the same format as done with acquire profiles. E.g. listing the available, directly followed by the contents of the profile, so the user immediately sees which profile does what. See line 21 and up in `acquire.rst` for the example.

That way, you avoid having to talk about the profile like this. Which currently, raises more questions for me reading it like this :)
